scylla_pg_lib/src/adapter.rs

```diff
@@ -62,16 +64,58 @@ pub trait DbExecute {
 #[async_trait]
 impl DbExecute for PgAdapter {
     async fn execute(&self, sql: &str, params: &[&(dyn ToSql + Sync)]) -> Result<Vec<Task>, PgAdapterError> {
-        let client: Client = self.pool.get().await?;
-        let stmt = client.prepare_cached(sql).await?;
-        let rows = client.query(&stmt, params).await?;
-        Ok(rows
-            .into_iter()
-            .map(|row| {
-                let task_value: Task = from_value(row.get(0)).unwrap();
-                task_value
-            })
-            .collect())
+        let max_tries = 5;
+        let mut try_count = 1;
+        let mut tasks: Option<Vec<Task>> = None;
+        let error: Option<PgAdapterError>;
+        loop {
+            let mut client: Client = self.pool.get().await?;
+            let stmt = client.prepare_cached(sql).await?;
+            let tx = client.build_transaction().isolation_level(IsolationLevel::RepeatableRead).start().await?;
+            match tx.query(&stmt, params).await {
+                Ok(rows) => {
+                    tasks = Some(
+                        rows.into_iter()
+                            .map(|row| {
+                                let task_value: Task = from_value(row.get(0)).unwrap();
+                                task_value
+                            })
+                            .collect(),
+                    );
+                    if let Err(e) = tx.commit().await {
+                        try_count += 1;
+                        log::error!("commit for tx failed : {}", e.to_string());
+                    } else {
+                        error = None;
+                        break;
+                    }
+                }
+                Err(e) => {
+                    if let Err(err) = tx.rollback().await {
```

Comment:
Don't need to explicitly `rollback`, it will be implicitly rolled backed when the transaction is dropped.